packages/trend_micro_vision_one/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,171 @@
+---
+description: Pipeline for processing Trend Micro Vision One Audit logs.
+processors:
+  - set:
+      field: ecs.version
+      value: '8.4.0'
+  - rename:
+      field: message
+      target_field: event.original
+      ignore_missing: true
+  - set:
+      field: event.kind
+      value: event
+  - json:
+      field: event.original
+      target_field: json
+      ignore_failure: true
+  - script:
+      description: Set the value of event.category and event.type.
+      lang: painless
+      source: >
+        def eventCategory = new HashSet();
+        def eventType = new HashSet();
+        if (ctx.json?.category != null && ctx.json.category != '' && ctx.json?.activity != null && ctx.json.activity != '') {
```

```suggestion
        if (ctx.json?.category != null && ctx.json.category != '' && ctx.json.activity != null && ctx.json.activity != '') {
```
The short circuit means that if `ctx.json` is null in the first expression, it will never be derefed in any of the later expressions.